src/app/games/hive/tests/HiveRules.spec.ts

```diff
@@ -0,0 +1,927 @@
+                [[b], [B]],
+            ];
+            const state: HiveState = HiveState.fromRepresentation(board, 2);
+
+            // When dropping a piece next to our piece
+            const move: HiveMove = HiveMove.drop(A, 2, 0).get();
+
+            // Then the move should succeed
+            const expectedBoard: Table<HivePiece[]> = [
+                [[b], [B], [A]],
+            ];
+            const expectedState: HiveState = HiveState.fromRepresentation(expectedBoard, 3);
+
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState);
+        });
+        it('should forbid dropping a piece adjacent to a piece of the opponent', () => {
+            // Given a state with a piece of the opponent already on the board
+            const board: Table<HivePiece[]> = [
+                [[B], [b]],
+            ];
+            const state: HiveState = HiveState.fromRepresentation(board, 2);
+
+            // When dropping a piece next to the opponent's piece
+            const move: HiveMove = HiveMove.drop(A, 2, 0).get();
+
+            // Then the move should fail
+            const reason: string = HiveFailure.CANNOT_DROP_NEXT_TO_OPPONENT();
+            RulesUtils.expectMoveFailure(rules, state, move, reason);
+        });
+        it('should forbid dropping a piece adjacent to a stack containing a piece of the opponent', () => {
+            // Given a state with a piece of the opponent already on the board, but under another piece
+            const board: Table<HivePiece[]> = [
+                [[B], [B, b]],
+            ];
+            const state: HiveState = HiveState.fromRepresentation(board, 2);
+
+            // When dropping a piece next to the stack
+            const move: HiveMove = HiveMove.drop(A, 2, 0).get();
+
+            // Then the move should fail
+            const reason: string = HiveFailure.CANNOT_DROP_NEXT_TO_OPPONENT();
+            RulesUtils.expectMoveFailure(rules, state, move, reason);
+        });
+        it('should allow dropping the queen bee at the fourth turn of a player', () => {
+            // Given a state in the fourth turn of player zero, without queen bee
+            const board: Table<HivePiece[]> = [
+                [[B], [G], [S], [b], [g], [s]],
+            ];
+            const state: HiveState = HiveState.fromRepresentation(board, 6);
+
+            // When dropping the queen bee
+            const move: HiveMove = HiveMove.drop(Q, 0, 1).get();
+
+            // Then the move should succeed
+            const expectedBoard: Table<HivePiece[]> = [
+                [[B], [G], [S], [b], [g], [s]],
+                [[Q], [], [], [], [], []],
+            ];
+            const expectedState: HiveState = HiveState.fromRepresentation(expectedBoard, 7);
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState);
+        });
+        it('should force dropping the queen bee by the fourth turn of a player (Player.ZERO)', () => {
```

Comment:
should force dropping the queen bee at turn 6 for Player.ZERO
et 
should force dropping the queen bee at turn 7 for Player.ONE ?